### ISSUE TITLE
Add link to gnomAD data

### ISSUE THREAD

**Issue by selinad** [ClinGen/clincoded#1291]
Provide link to gnomAD data for a variant -- e.g. http://gnomad.broadinstitute.org/variant/1-55516888-G-GA

If this URL difficult to build, we could just provide a link to the resource:

http://gnomad.broadinstitute.org/

Eventually, we will hopefully be able to get it from myvariant.info (need to discuss with them)

**Comment by jimmyzhen** [comment]
Hi @selinad,

I've implemented the gnomAD linkout as the following:
![image](https://cloud.githubusercontent.com/assets/6956310/22838805/95260168-ef7b-11e6-8dac-162e6aa2cd5b.png)

Let me know whether there is another location on the page more suitable for the display of this linkout. Thanks.

**Comment by selinad** [comment]
Hi @jimmyzhen -- wow, great!  So it will go straight to that allele in gnomAD -- what happens if there is no link?

@wrightmw do you think it should go separate from ExAC?  If so, would it deserve it's own blue bar and then there is a link below it?  Maybe after the ExAC data?

**Comment by selinad** [comment]
HI @jimmyzhen -- thx for the conversation with us.  Let us know what myvariant.info brings back for non-ExAC alleles.  For now, if we can only provide a link for alleles in ExAC, that is a great step forward.

Per our conversation with @wrightmw as well:

- Have gnomAD have it's own table and put "gnomAD" and variant representation (when available) in header (like ExAC).  No myvariant.info button since that's not yet possible.
- In body of table, maybe some descriptive text ("_gnomAD data is not currently available via API or download -- however, provided below are 2 links: 1) direct link to URL for variant in gnomAD (note: this may result in an error if no data is available in gnomAD for this variant) and 2) link to gnomAD's home page_"
- below note in body of table, the two links:
1) "Link to data for this variant in gnomAD" (note: this link will return an error if variant is not present in gnomAD)" (where the part in parentheses is not part of the link)
2) "Search gnomAD for this variant" ( @jimmyzhen please use same wording as we do for ExAC when we direct them to home page; also if we use variant in text, please)

@wrightmw please feel free to edit.

**Comment by jimmyzhen** [comment]
Hi @selinad,

Please review the requested changes in the following instance:
https://1291-jz-gnomad-link-8cfff53-jzhen.demo.clinicalgenome.org/

**Comment by selinad** [comment]
Hi @jimmyzhen great!  Thank you!

I'm going to explore some more variants, but it worked great for the one I tried.  
<img width="696" alt="screen shot 2017-02-11 at 7 46 06 am" src="https://cloud.githubusercontent.com/assets/11320314/22855047/3cf1453a-f02e-11e6-817f-8b93283f716f.png">

A couple of things:
- this text was not included after the "2 links":  "...  1) direct link to URL for variant in gnomAD (note: this may result in an error if no data is available in gnomAD for this variant) and 2) link to gnomAD's home page"
- after the first link, this text was not included "(note: this link will return an error if variant is not present in gnomAD)"

Not sure whether the first was just too wordy -- maybe we don't need it.  If we don't have the 2nd text above, however, we will need to adjust the link name as there is not necessarily data to be found when clicking on the link.  The 2 different links could go on separate rows if necessary.

thx for the quick implementation!

**Comment by selinad** [comment]
@jimmyzhen works great on more variants!

From our list of interesting variants, I found one that is not in ExAC (CV 41231).  Here is ExAC / gnomAD:
<img width="692" alt="screen shot 2017-02-11 at 7 56 25 am" src="https://cloud.githubusercontent.com/assets/11320314/22855142/c7e7fc64-f02f-11e6-9bd4-bb9b94018985.png">

Here is link to gnomAD for the variant:
<img width="1091" alt="screen shot 2017-02-11 at 7 56 14 am" src="https://cloud.githubusercontent.com/assets/11320314/22855145/d3edd902-f02f-11e6-890c-3da8a97e0155.png">

It looks like you were able to find the variant representation from myvariant.info for variants no in ExAC?  That is fantastic if the case.

Would you remind writing down how you were able to implement this based on our discussions, just so we make certain we know what to expect in testing ? (e.g. what myvariant.info brings back, etc. and the possible options returned based on the variant).  All the variants I looked at (many from the sheet - CA IDs, etc.) behaved perfectly.

thx @jimmyzhen for the quick implementation -- this is great and will be much appreciated.

**Comment by jimmyzhen** [comment]
Hi @selinad,

My implementation may be the optimal approach, and I welcome your feedback in this regard.

Here is the lowdown:
1. To construct the variant representation in a format used for the gnomAD URL, such as `16-11643576-G-A`, we essentially need the `chrom`, `pos`, `ref` and `alt` values of the variant.
2. In the myvariant.info response, I would be able to extract the described values above from different vendors, such as `exac` (if any), `clinvar` (if any), `cadd` or `vcf`.
3. Since the exac` data may not always be available in the myvariant.info response, I am using `clinvar` as the primary option to get the values in question. Then I fallback to `cadd` if `clinvar` is not present, and then fallback to `vcf` if neither `clinvar` nor `cadd` are present.
4. As long as the myvariant.info API call returns a `200 OK` status, we would be able to go into the JSON data and try to extract what we need from either one of the vendors, without being solely dependent on the `exac` data.
5. There are 2 scenarios in which we won't be able to get the needed values: 1) the myvariant.info API call returns a 404 Error status (no data found); 2) the variant is not a single nucleotide.

**Comment by selinad** [comment]
Thanks, @jimmyzhen !  If I followed all of this correctly, myvariant.info will return a status that reflects whether there will be a working link (for SNPs).  So, if it is not a SNP or a 200 OK from myvariant.info, we would not provide the direct link?  Let me know if this is correct and we can make certain the wording reflects everything correctly.  thx!

**Comment by jimmyzhen** [comment]
Hi @selinad,

You are correct. If it is not a SNP or a 200 OK from myvariant.info, we would not provide the direct link but rather a link to its homepage.

**Comment by jimmyzhen** [comment]
Hi @selinad,

I also think that the 1st instance of "(note:...)" seems a bit wordy and essentially repeating the 2nd instance of "(note:...)" appended to the 1st link. What do you think the following?
![image](https://cloud.githubusercontent.com/assets/6956310/22895874/d4ecc04e-f1d2-11e6-8915-a016a19c06b4.png)

**Comment by selinad** [comment]
HI @jimmyzhen -- here's what I understand:

- you may not always be able to produce a direct link
- if you do produce a direct link, it will go to a useful gnomAD page (variant not found, region not covered)
- there may be cases where you can't produce a direct link but gnomAD data is available.

Based on the above, how about this (explanatory text would be in italics):

_gnomAD data is not currently available via API or download; a direct link to gnomAD is provided whenever possible in addition to a link to gnomAD's home page_

Link to this variant in gnomAD
Search gnomAD

@wrightmw please feel free to edit!

**Comment by jimmyzhen** [comment]
Hi @selinad,

You can review the changes in the following screenshot:
![image](https://cloud.githubusercontent.com/assets/6956310/22911506/eee85476-f214-11e6-8a25-1957fa52924d.png)

**Comment by selinad** [comment]
love it!  Thanks, @jimmyzhen.  If simple could we add the word "however" (my bad) -- like this:

_gnomAD data is not currently available via API or download; however, a direct link to gnomAD is provided whenever possible in addition to a link to gnomAD's home page_

**Comment by jimmyzhen** [comment]
Updated:
![image](https://cloud.githubusercontent.com/assets/6956310/22911778/59b915fa-f216-11e6-8399-d5a4d59f6d9f.png)

**Comment by selinad** [comment]
Super!

**Comment by selinad** [comment]
Nice @jimmyzhen -- works well for variants where there is no gnomAD data even when you can get link and for variant (21135) where there is data.  

I did have a problem for this variant: CV13346 -- it says "Data is currently only returned for single nucleotide variants. Search gnomAD" but the link goes here instead of home page:

http://gnomad.broadinstitute.org/variant/12-112888163-GTG--

This should be checked for all variants where a link can't be built.  Not a show stopper, but gives an error and is incorrect link.

**Comment by wrightmw** [comment]
Another minor thing on this one. The text is different for Gnomad for the same functionality (i.e. diverting to homepage for non-SNVs) compared to the other population databases. They all say "for this variant." at the end:
![screen shot 2017-02-16 at 5 40 52 pm](https://cloud.githubusercontent.com/assets/15131169/23049219/806a5ab4-f46f-11e6-862c-c4da56a58990.png)

**Comment by wrightmw** [comment]
Other non-SNV variants for testing
Duplication: VariationID: 7185, VariationID: 53905
Deletion: VariationID: 53237, VariationID: 53205
Insertion: VariationID: 54032
Indel: VariationID: 53981

**Comment by wrightmw** [comment]
Bug ticket created #1302

**Comment by wrightmw** [comment]
Text  change ticket added #1303

**Comment by kilodalton** [comment]
Included in last release (R10). Nice job and thanks for your hard work.
